addons/website_slides_survey/models/survey_user.py

```diff
@@ -11,3 +11,40 @@ class SurveyUserInput(models.Model):
         help="The related course slide when there is no membership information")
     slide_partner_id = fields.Many2one('slide.slide.partner', 'Subscriber information',
         help="Slide membership information for the logged in user")
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        records = super(SurveyUserInput, self).create(vals_list)
+        records._check_for_failed_attempt()
+        return records
+
+    @api.multi
+    def write(self, vals):
+        res = super(SurveyUserInput, self).write(vals)
+        self._check_for_failed_attempt()
```

Comment:
Shouldn't we do the check only if some fields are modified ? Like if we have state, quiz_passed, ... in vals ? Purpose is to avoid recomputing too much stuff.

Reply:
Good point. It's done.